src/op_handlers/ret.rs

```diff
@@ -63,7 +63,7 @@ pub fn ret(
         vm.clear_registers();
         vm.set_register(1, result);
         let previous_frame = vm.pop_frame()?;
-        vm.current_frame_mut()?.gas_left += previous_frame.gas_left;
+        vm.current_frame_mut()?.gas_left += previous_frame.gas_left.0 - previous_frame.stipend;
```

Comment:
Let's use the decrease_gas method here.
There's nothing wrong with exposing the zero field, but we must do increase and decrease of gas with their corresponding methods to avoid over/underflows.

Reply:
Maybe we could make it a private member instead to avoid mistakes.

Reply:
It would be a good idea to make it private. Not for this pr though, I think there are a lot of places that would require a refactor.